File: EpicGame.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    int a, b, n;
    cin >> a >> b >> n;
    int ans = 0;
    bool turn = true;
    while (n>0) {
        int temp;
        if (turn) {
            temp = __gcd(a, n);
            n -= temp;
            turn = false;
            if (temp>n)
                ans = 0;
        }
        else {
            temp = __gcd(b, n);
            n -= temp;
            turn = true;
            if (temp>n)
                ans = 1;
        }
    }
    cout << ans << endl;

    return 0;
}

File: Python/EpicGame.py
import math


def main():
    a = int(input())
    b = int(input())
    n = int(input())
    ans = 0
    turn = True
    while n > 0:
        temp = 0
        if turn:
            temp = math.gcd(a, b)
            n -= temp
            turn = False
            if temp > n:
                ans = 0
        else:
            temp = math.gcd(b, n)
            n -= temp
            turn = True
            if temp > n:
                ans = 1
    print(ans)


if __name__ == '__main__':
    main()
